models/DataObject/ClassDefinition/Data.php

```diff
@@ -404,6 +404,12 @@ public function getFilterConditionExt(mixed $value, string $operator, array $par
             return $key . ' ' . $operator . ' ' . $value . ' ' . $trailer;
         }
 
+        if ($operator === 'in') {
+            $formattedValues = implode(',', array_map('floatval', explode(',', $value)));
```

Comment:
Or maybe even better:
```suggestion
            $formattedValues = implode(',', array_map(floatval(...), explode(',', $value)));
```

Reply:
yes even better!